src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -619,37 +547,13 @@ void registerBlobs(List<BlobMetadata> blobs, final int executionCount) {
         this.name,
         executionCount);
 
-    RegisterBlobResponse response = null;
-    try {
-      Map<Object, Object> payload = new HashMap<>();
-      payload.put(
-          "request_id", this.flushService.getClientPrefix() + "_" + counter.getAndIncrement());
-      payload.put("blobs", blobs);
-      payload.put("role", this.role);
-
-      response =
-          executeWithRetries(
-              RegisterBlobResponse.class,
-              REGISTER_BLOB_ENDPOINT,
-              payload,
-              "register blob",
-              STREAMING_REGISTER_BLOB,
-              httpClient,
-              requestBuilder);
-
-      // Check for Snowflake specific response code
-      if (response.getStatusCode() != RESPONSE_SUCCESS) {
-        logger.logDebug(
-            "Register blob request failed for blob={}, client={}, message={}, executionCount={}",
-            blobs.stream().map(BlobMetadata::getPath).collect(Collectors.toList()),
-            this.name,
-            response.getMessage(),
-            executionCount);
-        throw new SFException(ErrorCode.REGISTER_BLOB_FAILURE, response.getMessage());
-      }
-    } catch (IOException | IngestResponseException e) {
-      throw new SFException(e, ErrorCode.REGISTER_BLOB_FAILURE, e.getMessage());
-    }
+    RegisterBlobRequest request =
+        new RegisterBlobRequest(
+            this.storageManager.getClientPrefix() + "_" + counter.getAndIncrement(),
```

Comment:
nit: can we move counter inside serviceClient, instead of making all the API callsites do the requestId computation and getAndIncrement ? ok to do in a future PR / as a P2.

Reply:
Added as backlog in iceberg file path generation.